### DISCUSSION TITLE
How to display a consumed library version in storybook composition UI

### DISCUSSION THREAD

**Discussion by lawlesscreation** [storybookjs/storybook#20871] [Help]
I'm trying to find a way to show the version of a consumed storybook library when using storybook composition... something like this:

<img width="247" alt="Screenshot 2023-02-01 at 12 34 01" src="https://user-images.githubusercontent.com/299914/216043751-515ddc72-fd01-4463-9737-5380495f925a.png">

But I'm trying to get the version *from* the library itself and have storybook composition read that version number.

I've tried a few things:

- attempted to set it in the main.js file, but this happens at build-time so I cannot access anything in the `window`
```
// main.js
return {
        library: {
          title: `Library (v???)`,
          url: 'http://my-storybook-library-url.com'
        }
```
- I thought the metadata.json would be the right way to do it, but that seems to be for providing a list of versions - not the actual version you're consuming.
- Thought about adding a custom script to storybook or using manager.js to fetch a custom metadata file and perhaps append the version to the sidebar but that sounds a little hacky.

Perhaps I'm missing something.

In essence, what I'm trying to do is for the library to say what version *it* is and for composition to display that version number in the sidebar.

Can anyone suggest an easy way to do that? Thanks.

**Comment by lawlesscreation** [comment]
Currently, the only way I can find to do this is with manager.js:

```
window.addEventListener('load', (event) => { // have to wait for load, otherwise the change is removed when the stories finish loading!?
  fetch('path-to-custom-metadata.json')
    .then(response => response.json())
    .then(data => {
      const title = document.getElementById('storybook-explorer-tree').querySelector('[title="library"]');
      title.insertAdjacentHTML('afterend', `<span>${data.version}</span>`);
  });
})
```

Very hacky :)

**Comment by groteck** [comment]
I used the titlePrefix in the config:
![image](https://github.com/storybookjs/storybook/assets/833030/db7858a3-e67c-4769-8638-6a00fc18ed6e)
With this it displays the version like:
![image](https://github.com/storybookjs/storybook/assets/833030/66591ebe-1cc7-4ad6-b7e3-4c6b9650da86)
